creator-node/src/routes/nodeSync.js

```diff
@@ -67,6 +68,19 @@ module.exports = function (app) {
       const ipfs = req.app.get('ipfsAPI')
       let ipfsIDObj = await getIPFSPeerId(ipfs, config)
 
+      // Ensure all relevant files are available through IPFS at export time
+      await Promise.all(files.map(async (file) => {
+        if (file.type === 'track' || file.type === 'metadata' || file.type === 'copy320') {
+          await rehydrateIpfsFromFsIfNecessary(ipfs, file.multihash, file.storagePath)
+        } else if (file.type === 'image') {
+          await rehydrateIpfsFromFsIfNecessary(
```

Comment:
@dmanjunath thoughts on this here?

Reply:
@hareeshnagaraj i think we have to try to rehydrate here. otherwise the files may not be available on ipfs even if they're available on disk. now the one case we should try to make sure we handle is if a file is not available on disk. the only time I can think of that happening is like this


user uploads a track to primary
backups start syncing and export function runs, but primary goes down before all files synced to backup and available on backups ipfs and file system
backup 1 becomes primary
new backup is assigned and starts syncing from new primary, but some files don't exist on the file system

Reply:
also i don't think you need the async and await in from of `async (file) => .... await rehydrateIpfsFromFsIfNecessary`.  `rehydrateIpfsFromFsIfNecessary` returns a promise anyway so the await Promise.all() should catch that

Reply:
so is there the possibility of adding a hard catch to 'primary going down'? like on shutdown of the express app, ensure that everything is exported still? bc we rarely see a _hard_ crash like SIGKILL or something, usually the app does shut down gracefully

Reply:
@hareeshnagaraj what do you mean exported still? exported where?

Reply:
like re-added to ipfs if missing, basically the new 'export' code that rehydrates everything